frontend/server/src/Controllers/User.php

```diff
@@ -2105,11 +2124,35 @@ public static function apiStats(\OmegaUp\Request $r): array {
 
         return [
             'runs' => \OmegaUp\DAO\Runs::countRunsOfIdentityPerDatePerVerdict(
-                intval($identity->identity_id)
+                $identity->identity_id
             ),
         ];
     }
 
+    /**
+     * It will return a boolean value indicating whether user can see
+     * information of the given identity. There are three scenarios where this
+     * function returns false:
+     *  - Logged user is a sysadmin
+     *  - Logged user is trying to see its own information
+     *  - Identity whose logged user wants to see its information is not private
+     */
+    private static function shouldUserInformationBeHidden(
+        ?\OmegaUp\DAO\VO\Identities $loggedIdentity,
+        \OmegaUp\DAO\VO\Identities $identity,
+        ?\OmegaUp\DAO\VO\Users $user
+    ): bool {
+        return (
+            is_null($loggedIdentity)
+            || (
+                $loggedIdentity->username != $identity->username
+                && !\OmegaUp\Authorization::isSystemAdmin($loggedIdentity)
+            )
+        )
+        && (!is_null($user)
+        && $user->is_private == 1);
```

Comment:
```suggestion
        && !is_null($user)
        && $user->is_private == 1;
```

así es un poquito más fácil identificar que estas dos cláusulas van "en el mismo nivel".

Reply:
Listo.